### PR TITLE
Python3

### PR DESCRIPTION
**Pull request by videlec**
This branch is based on top of doctests. I had to make a little hack to make the test pass, namely providing the option `IGNORE_EXCEPTION_DETAIL` to doctest. There is something fishy going on here.

### CODE REVIEW

File: cypari2/closure.pyx [hunk 1]
```diff
@@ -133,39 +135,40 @@ cpdef Gen objtoclosure(f):
...
-        ...
-        PariError: call_python: too many parameters in user-defined function call
-        sage: mul([1], [2])
-        Traceback (most recent call last):
-        ...
-        PariError: call_python: forbidden multiplication t_VEC (1 elts) * t_VEC (1 elts)
+    Examples:
+
+    >>> from cypari2.closure import objtoclosure
+    >>> def pymul(i,j): return i*j
```
Comment (defeo) [inline cypari2/closure.pyx]: Why this change? I think that was meant **explicitly** to test lambdas.

File: cypari2/convert.pyx [hunk 2]
```diff
@@ -104,63 +115,70 @@ cpdef gen_to_integer(Gen x):
...
+    1
+    >>> gen_to_integer(pari("3/2 + 0.0*I"))
+    1
+    >>> gen_to_integer(pari("Mod(-1, 11)"))
+    10
+    >>> gen_to_integer(pari("5 + O(5^10)"))
+    5
+    >>> gen_to_integer(pari("Pol(42)"))
+    42
+    >>> gen_to_integer(pari("u"))
```
Comment (defeo) [inline cypari2/convert.pyx]: So, did you get to the bottom of this? Why was it failing on Travis?

File: cypari2/convert.pyx [hunk 3]
```diff
@@ -589,7 +615,7 @@ cpdef gen_to_python(Gen z):
     elif t == t_VEC or t == t_COL:
         return [gen_to_python(x) for x in z.python_list()]
     elif t == t_VECSMALL:
-        return z.python_list_small()
+       return z.python_list_small()
```
Comment (defeo) [inline cypari2/convert.pyx]: This diff is weird!

File: cypari2/pari_instance.pyx [hunk 4]
```diff
@@ -616,14 +589,6 @@ cdef class Pari(Pari_auto):
         Print the internal PARI variables ``top`` (top of stack), ``avma``
         (available memory address, think of this as the stack pointer),
         ``bot`` (bottom of stack).
-
-        EXAMPLES::
-
-            sage: pari.debugstack()  # random
```
Comment (defeo) [inline cypari2/pari_instance.pyx]: Isn't there a way to leave the example undoctested?

File: cypari2/closure.pyx (hunk 1 above)
Comment (videlec) [inline cypari2/closure.pyx]: lambda functions behavior are different between Python2 and Python3. In Python3, a lambda function has only one argument.

File: cypari2/convert.pyx (hunk 2 above)
Comment (videlec) [inline cypari2/convert.pyx]: No idea. This is still failing on travis. However the option `IGNORE_EXCEPTION_DETAIL` in doctests make this failure disappear.

File: cypari2/convert.pyx (hunk 3 above)
Comment (videlec) [inline cypari2/convert.pyx]: Indeed! Fixed  by d7b7a3e

File: cypari2/pari_instance.pyx (hunk 4 above)
Comment (videlec) [inline cypari2/pari_instance.pyx]: No without a hook to doctest. This is what they did in cypari and what we might want to do... (they also have a hook for 32 vs 64 bits doctests)

File: cypari2/closure.pyx (hunk 1 above)
Comment (defeo) [inline cypari2/closure.pyx]: ?? What do you mean? I'm using Python 3.6, and I see no problem with the old test

```
>>> objtoclosure(lambda i,j: i*j)
(v1,v2,v3,v4,v5)->call_python(v1,v2,v3,v4,v5,140180350752424)
```

File: cypari2/convert.pyx (hunk 2 above)
Comment (defeo) [inline cypari2/convert.pyx]: We may at least want to open an issue for this, or we may forget about it.

File: cypari2/pari_instance.pyx (hunk 4 above)
Comment (defeo) [inline cypari2/pari_instance.pyx]: All right, we can think about this later.

File: cypari2/closure.pyx (hunk 1 above)
Comment (videlec) [inline cypari2/closure.pyx]: Indeed. I do not know what happend in the first place. This is back with `418cf40`.

File: cypari2/convert.pyx (hunk 2 above)
Comment (videlec) [inline cypari2/convert.pyx]: #14

**Comment by jdemeyer** [comment]
It's wrong to assume that every string coming from PARI is ASCII. I mean, people should be able to write PARI/GP code like `print("Author: Jean-François Mestre")`.
When in doubt, use [sys.getfilesystemencoding()](https://docs.python.org/3/library/sys.html#sys.getfilesystemencoding) which is what Python uses by default.

I also don't agree with changing `<bytes?>` to `<bytes>`: it's an extra safety check.

**Comment by jdemeyer** [comment]
This is also wrong, because you aren't calling `str()` anymore:
```
{name} = {name}.encode('ascii') if isinstance({name}, str) else bytes({name})
```

This should be something like
```
{name} = str({name})
if PYTHON3:
    {name} = {name}.encode(filesystemencoding)
```

**Comment by videlec** [comment]
Changes

- removing `system.pxi` and use `compile_time_env` in cythonization
- functions `to_bytes` and `to_string` to avoid repetitions

Travis looks

**Comment by videlec** [comment]
ping!

**Comment by jdemeyer** [comment]
I don't think it is good style to hard-code the Python version number in the `.c` sources. Better make it a run-time check.

**Comment by jdemeyer** [comment]
Some more docs for `to_string` would be welcome. What do you mean with "string" in this context? Do you mean the `str` type (which is `bytes` in Python 2 and `unicode` in Python 3)?

If so, it seems more reasonable to have `to_bytes` and `to_unicode` functions, where `to_string = to_bytes` on Python 2 and `to_string = to_unicode` on Python 3. That would simplify things.

Also: what is acceptable input for `to_string`? Both `bytes` and `unicode` on Python 2 and Python 3?

**Comment by videlec** [comment]
@jdemeyer concerning your harceded version, what is the point? and what are you referring to exactly? Note that only the major version (2 or 3) is taken into account. And a Python2 compiled version will not be Python3 compatible. And what is happening is during cythonization, it is not hardcoded in the compiled version.

**Comment by jdemeyer** [comment]
> concerning your harceded version, what is the point?

You make the assumption that the version of Python which is used by Cython equals the version of Python which will use cypari2. There is no need to make that assumption.

Not making this assumption could allow to install cypari2 on systems which do not have Cython installed.

**Comment by videlec** [comment]
>>    concerning your harceded version, what is the point?
>
> You make the assumption that the version of Python which is used by Cython equals the version of Python which will use cypari2. There is no need to make that assumption.

Agreed. But the target Python version has to be known at compilation time anyway (e.g. `__cmp__` should not be there in Python3). How would you pass such parameter to the setup script?

> Not making this assumption could allow to install cypari2 on systems which do not have Cython installed.

Then you would need two versions of the C files. One Python2 compatible and one Python3 compatible.

I don't think this point is relevant to the current PR.

**Comment by jdemeyer** [comment]
> e.g. __cmp__ should not be there in Python3

That's for Cython to decide, not for us. If you write `def __cmp__` in an extension type, Cython knows to ignore that on Python 3.

Cython tries hard to output `.c` code that is compatible with both Python 2 and Python 3. Let's not destroy that by hard-coding the Python version number in those conversion functions.

**Comment by videlec** [comment]
Funny cython behavior... with

    IF PY_MAJOR_VERSION == 2:
        return to_bytes(s)
    ELSE:
        return to_unicode(s)

I got with python2

    Error compiling Cython file:
    ------------------------------------------------------------
    ...
        True
        >>> s1 == s2 == s3 == 'hello'
        True
        """
        IF PY_MAJOR_VERSION == 2:
            return to_bytes(s)
                          ^
    ------------------------------------------------------------    
    cypari2/string_utils.pyx:71:23: Cannot convert 'bytes' object to str implicitly. This is not portable to Py3.

and with Python3

    Error compiling Cython file:
    ------------------------------------------------------------
    ...
        True
        """
        IF PY_MAJOR_VERSION == 2:
            return to_bytes(s)
        ELSE:
            return to_unicode(s)
                            ^
    ------------------------------------------------------------    
    cypari2/string_utils.pyx:73:25: Cannot convert Unicode string to 'str' implicitly. This is not portable and requires explicit encoding.
    Traceback (most recent call last):

**Comment by videlec** [comment]
>>   e.g. cmp should not be there in Python3
>
> That's for Cython to decide, not for us. If you write
> def __cmp__ in an extension type, Cython knows to
> ignore that on Python 3.
>
> Cython tries hard to output .c code that is compatible
> with both Python 2 and Python 3. Let's not destroy that
> by hard-coding the Python version number in those
> conversion functions.

Let me repeat that it is not hardcoded anywhere. There is a constant Cython macro that enables/disables some portion of the code... and it is trivial to make it configurable.

**Comment by jdemeyer** [comment]
> Let me repeat that it is not hardcoded anywhere

It is hardcoded in the `.c` file. Let me try to explain in more detail:

The process of compilation is: `.pyx -> .c -> .so`

The `.so` file obviously has to depend on the Python version since it makes calls into the Python C/API, so it needs to link to the Python library.

But the `.c` file does not need to depend on the Python version. Normally, Cython produces a `.c` file which works in Python 2 and Python 3. By passing `PY_MAJOR_VERSION` to Cython, you are forcing the `.c` file to depend on the Python version used by Cython. There is no reason to do that.

**Comment by videlec** [comment]
I see. The main problem is actually about doctests because in Python3

    NameError: name 'long' is not defined
    NameError: name 'cmp' is not defined

How would you test `cmp(pari(1), pari(2))` and `log(pari(3))` in Python2 but disable these tests in Python3?

**Comment by jdemeyer** [comment]
I guess you will need to look at `sys.version_info` in the doctest.

**Comment by videlec** [comment]
... will look horrible, but at least it is doable

**Comment by videlec** [comment]
All right, done! No more compile time env.

**Comment by jdemeyer** [comment]
Just to be safe, replace `PY_MAJOR_VERSION == 2` by `PY_MAJOR_VERSION <= 2` and `elif PY_MAJOR_VERSION == 3` by `else` (unless you expect that Python 4 will use yet another string type).

**Comment by jdemeyer** [comment]
It's probably more efficient to change
```
elif isinstance(s, unicode):
        return s
```
to
```
elif isinstance(s, unicode):
        return <unicode>s
```
and analogous for `to_bytes`.

(unless Cython is clever enough to infer this because of the `isinstance` check?)

**Comment by jdemeyer** [comment]
You can simplify
```
    if PY_MAJOR_VERSION == 2:
        if isinstance(x, int):
            A
        elif isinstance(x, long):
            B
    elif PY_MAJOR_VERSION == 3:
        if isinstance(x, int):
            B
```
by
```
        if isinstance(x, long):
            B
        elif isinstance(x, int):
            A
```
In "Cython 3", `int` is the same as `long` so this code would do the right thing on Python 2 and Python 3.

**Comment by videlec** [comment]
sadly network in Bordeaux is down... so Travis is not happy.

**Comment by videlec** [comment]
network back, Travis happy!

File: cypari2/convert.pyx [hunk 5]
```diff
@@ -145,14 +150,17 @@ cpdef gen_to_integer(Gen x):
...
-    >>> gen_to_integer(pari("1.0 - 2^64"))
-    -18446744073709551615L
-    >>> gen_to_integer(pari("1 - 2^64"))
-    -18446744073709551615L
+    >>> gen_to_integer(pari("1.0 - 2^64")) == -18446744073709551615
+    True
+    >>> gen_to_integer(pari("1 - 2^64")) == -18446744073709551615
+    True
+    >>> import sys
+    >>> if sys.version_info.major == 2:
```
Comment (jdemeyer) [inline cypari2/convert.pyx]: The same test should be done with `int` instead of `long` (and you might as well do it on Python 2 and Python 3).

File: cypari2/convert.pyx [hunk 6]
```diff
@@ -525,21 +535,21 @@ cpdef gen_to_python(Gen z):
     [1, 2, 3]
     >>> type(a1)
     <... 'list'>
-    >>> map(type, a1)
+    >>> list(map(type, a1))
```
Comment (jdemeyer) [inline cypari2/convert.pyx]: This is ugly. Write `[type(x) for x in a1]`.

File: cypari2/convert.pyx [hunk 7]
```diff
@@ -525,21 +535,21 @@ cpdef gen_to_python(Gen z):
...
     [<... 'int'>, <... 'int'>, <... 'int'>]
 
     >>> a2 = gen_to_python(z2); a2
     [1, 3.4, [-5, 2], inf]
     >>> type(a2)
     <... 'list'>
-    >>> map(type, a2)
+    >>> list(map(type, a2))
```
Comment (jdemeyer) [inline cypari2/convert.pyx]: Idem

File: cypari2/convert.pyx [hunk 8]
```diff
@@ -525,21 +535,21 @@ cpdef gen_to_python(Gen z):
...
     [<... 'int'>, <... 'float'>, <... 'list'>, <... 'float'>]
 
     >>> a3 = gen_to_python(z3); a3
     [1, 5.2]
     >>> type(a3)
     <... 'list'>
-    >>> map(type, a3)
+    >>> list(map(type, a3))
```
Comment (jdemeyer) [inline cypari2/convert.pyx]: Idem

File: cypari2/convert.pyx [hunk 9]
```diff
@@ -607,6 +617,6 @@ cpdef gen_to_python(Gen z):
         else:
             return -INFINITY
     elif t == t_STR:
-        return str(z)
+        return to_string(<bytes> GSTR(g))
```
Comment (jdemeyer) [inline cypari2/convert.pyx]: Is the `<bytes>` really needed here? (I haven't checked, maybe it is)

File: cypari2/gen.pyx [hunk 10]
```diff
@@ -38,6 +38,9 @@ AUTHORS:
```
Comment (jdemeyer) [inline cypari2/gen.pyx]: Given that you rely on *explicit* conversions in many other places, I think it's better to be explicit everywhere. This means dropping `cython: c_string_encoding=default` and adding `to_string` or `to_bytes` where needed.

File: cypari2/gen.pyx [hunk 11]
```diff
@@ -1782,25 +1801,20 @@ cdef class Gen(Gen_auto):
...
-        >>> long(pari(2**31-1))
-        2147483647L
-        >>> long(pari(-2**31))
-        -2147483648L
-        >>> long(pari("Pol(10)"))
-        10L
-        >>> long(pari("Mod(2, 7)"))
-        2L
+        >>> if sys.version_info.major == 3:
+        ...     long = int
```
Comment (jdemeyer) [inline cypari2/gen.pyx]: +1

File: cypari2/gen.pyx [hunk 12]
```diff
@@ -4182,8 +4196,8 @@ cdef class Gen(Gen_auto):
                 return new_gen(gsubst(self.g, varn(self.g), t0.g))
 
         # Call substvec() using **kwds
-        vstr = kwds.keys()            # Variables as Python strings
-        t0 = objtogen(kwds.values())  # Replacements
+        vstr = list(kwds.keys())            # Variables as Python strings
+        t0 = objtogen(list(kwds.values()))  # Replacements
```
Comment (jdemeyer) [inline cypari2/gen.pyx]: `list()` should not be needed here: `objtogen` can deal with generators.

File: cypari2/pari_instance.pyx [hunk 13]
```diff
@@ -385,22 +387,22 @@ def prec_words_to_dec(long prec_in_words):
...
-cdef PariOUT sage_pariOut
+cdef PariOUT python_pariOut
 
-cdef void sage_putchar(char c):
+cdef void python_putchar(char c):
     cdef char s[2]
     s[0] = c
     s[1] = 0
-    sys.stdout.write(s)
+    sys.stdout.write(to_string(<bytes> s))
```
Comment (jdemeyer) [inline cypari2/pari_instance.pyx]: I don't like this since you are converting `bytes` to `str` which `sys.stdout` immediately needs to convert back to `bytes`. In Python 3, you can use `sys.stdout.buffer` for the underlying stream which accepts `bytes`. So you could do something like:
```
try:
    stdout_bytes = sys.stdout.bytes
except AttributeError:
    stdout_bytes = sys.stdout
stdout_bytes.write(s)
```

File: tests/rundoctest.py [hunk 14]
```diff
@@ -1,17 +1,23 @@
...
 attempted = 0
 for mod in [cypari2.closure, cypari2.convert, cypari2.gen,
-            cypari2.handle_error, cypari2.pari_instance, cypari2.stack]:
+            cypari2.handle_error, cypari2.pari_instance, cypari2.stack,
+            cypari2.string_utils]:
 
     print("="*80)
     print("Testing {}".format(mod.__name__))
-    test = doctest.testmod(mod, optionflags=doctest.ELLIPSIS|doctest.REPORT_NDIFF)
+    test = doctest.testmod(mod, optionflags=doctest.ELLIPSIS|doctest.REPORT_NDIFF|doctest.IGNORE_EXCEPTION_DETAIL)
```
Comment (jdemeyer) [inline tests/rundoctest.py]: Is this still needed?

File: cypari2/convert.pyx (hunk 5 above)
Comment (videlec) [inline cypari2/convert.pyx]: see 002da53

File: cypari2/convert.pyx (hunk 6 above)
Comment (videlec) [inline cypari2/convert.pyx]: see 15901f9

File: cypari2/convert.pyx (hunk 7 above)
Comment (videlec) [inline cypari2/convert.pyx]: also 15901f9

File: cypari2/convert.pyx (hunk 8 above)
Comment (videlec) [inline cypari2/convert.pyx]: also 15901f9

File: cypari2/convert.pyx (hunk 9 above)
Comment (videlec) [inline cypari2/convert.pyx]: It is indeed not needed (see 92f558b)

File: cypari2/gen.pyx (hunk 10 above)
Comment (videlec) [inline cypari2/gen.pyx]: see b5e5375

File: cypari2/gen.pyx (hunk 12 above)
Comment (videlec) [inline cypari2/gen.pyx]: see fa24efb

File: cypari2/pari_instance.pyx (hunk 13 above)
Comment (videlec) [inline cypari2/pari_instance.pyx]: Does not work as is since sys.stdout is sometimes redirected (e.g. doctest). I used

    try:
        sys.stdout.buffer.write(s)
    except AttributeError:
        sys.stdout.write(to_string(s))

See b4410fc

File: tests/rundoctest.py (hunk 14 above)
Comment (videlec) [inline tests/rundoctest.py]: It is. In Python3, pari error are printed

    Traceback (most recent call last):
    ...
    cypari2.handle_error.PariError: MSG

while in Python2 it is

    Traceback (most recent call last):
    ...
    PariError: MSG

File: cypari2/pari_instance.pyx [hunk 15]
```diff
@@ -385,22 +387,30 @@ def prec_words_to_dec(long prec_in_words):
...
+cdef void python_putchar(char c):
     cdef char s[2]
     s[0] = c
     s[1] = 0
-    sys.stdout.write(s)
+    try:
+        # avoid string conversion if possible
+        sys.stdout.buffer.write(s)
+    except AttributeError:
+        sys.stdout.write(to_string(s))
```
Comment (jdemeyer) [inline cypari2/pari_instance.pyx]: You don't need `to_string()` here. The whole point is that you want to write `bytes` without conversion.

File: cypari2/pari_instance.pyx [hunk 16]
```diff
@@ -385,22 +387,30 @@ def prec_words_to_dec(long prec_in_words):
...
     pari_set_last_newline(1)
 
-cdef void sage_puts(const char* s):
-    sys.stdout.write(s)
+cdef void python_puts(const char* s):
+    try:
+        # avoid string conversion if possible
+        sys.stdout.buffer.write(s)
+    except AttributeError:
+        sys.stdout.write(to_string(s))
```
Comment (jdemeyer) [inline cypari2/pari_instance.pyx]: You don't need `to_string()` here. The whole point is that you want to write `bytes` without conversion.

File: cypari2/gen.pyx [hunk 17]
```diff
@@ -4182,14 +4193,14 @@ cdef class Gen(Gen_auto):
                 return new_gen(gsubst(self.g, varn(self.g), t0.g))
 
         # Call substvec() using **kwds
-        vstr = kwds.keys()            # Variables as Python strings
-        t0 = objtogen(kwds.values())  # Replacements
+        vstr = iter(kwds.keys())            # Variables as Python strings
```
Comment (jdemeyer) [inline cypari2/gen.pyx]: Should be `kwds.iterkeys()`

File: cypari2/string_utils.pyx [hunk 18]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+r"""
+Conversion functions for bytes/unicode
+"""
+
+import sys
+encoding = sys.getfilesystemencoding()
+cdef int PY_MAJOR_VERSION = sys.version_info.major
```
Comment (jdemeyer) [inline cypari2/string_utils.pyx]: It would be more efficient to make this a compile-time constant with
```
cdef extern from *:
   int PY_MAJOR_VERSION
```

File: cypari2/string_utils.pyx [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+    True
+    """
+    if isinstance(s, bytes):
+        return <bytes> s
+    elif isinstance(s, unicode):
+        return (<unicode> s).encode(encoding)
+    else:
+        raise TypeError
+
+cpdef unicode to_unicode(s):
```
Comment (jdemeyer) [inline cypari2/string_utils.pyx]: Maybe `cpdef inline` for efficiency.

File: cypari2/string_utils.pyx [hunk 20]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+r"""
+Conversion functions for bytes/unicode
+"""
+
+import sys
+encoding = sys.getfilesystemencoding()
+cdef int PY_MAJOR_VERSION = sys.version_info.major
+
+cpdef bytes to_bytes(s):
```
Comment (jdemeyer) [inline cypari2/string_utils.pyx]: Maybe `cpdef inline` for efficiency.

File: cypari2/pari_instance.pyx (hunk 15 above)
Comment (videlec) [inline cypari2/pari_instance.pyx]: Yes you do. It is perfectly valid to redirect `sys.stdout` to some other streams that might not support the buffer thing (like the doctest module does). In other words the `except` is not only intended for Python2. In particular the `to_string` is mandatory in Python3 (and useless in Python2). Note that `to_string` does not perform any conversion in Python2.

File: cypari2/gen.pyx (hunk 17 above)
Comment (videlec) [inline cypari2/gen.pyx]: `iterkeys` does not exist in Python3: `kwds.keys()` is already an iterator.

File: cypari2/string_utils.pyx (hunk 20 above)
Comment (videlec) [inline cypari2/string_utils.pyx]: I thought it was wrong to declare an inline function inside a .pyx that is used in other parts of the code (here in auto_instance.pxi)

File: cypari2/string_utils.pyx (hunk 18 above)
Comment (videlec) [inline cypari2/string_utils.pyx]: done in d9efdf8

**Comment by jdemeyer** [comment]
> iterkeys does not exist in Python3:

It does exist in Cython, which is what matters here.

**Comment by jdemeyer** [comment]
> It is perfectly valid to redirect sys.stdout to some other streams that might not support the buffer thing (like the doctest module does).

The question is: can it happen that `sys.stdout` does **not** support the buffer thing and that it also does **not** accept `bytes`?

**Comment by jdemeyer** [comment]
> I thought it was wrong to declare an inline function inside a .pyx

It is wrong to **declare** it as inline in a **.pxd** file. But it's fine to  **define** an inline function in a **.pyx** file.

**Comment by jdemeyer** [comment]
> inlined to_string

That's not what I meant, but that's actually a good solution too.

**Comment by jdemeyer** [comment]
Not `iter(vstr.iterkeys())` but just `vstr.iterkeys()`.

**Comment by videlec** [comment]
> Not iter(vstr.iterkeys()) but just vstr.iterkeys().

does not work

    TypeError: dict_keys object is not an iterator

**Comment by videlec** [comment]
>>    It is perfectly valid to redirect sys.stdout to some other streams that might not support the buffer thing (like the doctest module does).

> The question is: can it happen that sys.stdout does not support the buffer thing and that it also does not accept bytes?

Removing the `to_string`, while running the doctests in Python 3 you get hundreds of

    Exception ignored in: 'cypari2.pari_instance.python_puts'
    TypeError: string argument expected, got 'bytes'

**Comment by jdemeyer** [comment]
> Removing the to_string, while running the doctests in Python 3

I see. This is probably a doctest-specific thing.

**Comment by jdemeyer** [comment]
> does not work (at least in Python 3)
> 
> TypeError: dict_keys object is not an iterator

You're right. I though that `dict.keys()` in Python 3 returned the same thing as `dict.iterkeys()` in Python 2, but apparently, that's not the case. You get an **iterable** but not an **iterator**.

**Comment by jdemeyer** [comment]
OK, this is a mess but I guess it's the best we can do. For me, there are no further changes needed.

**Comment by videlec** [comment]
@defeo do you have any comment?

**Comment by videlec** [comment]
@defeo Let say that 6 days mean yes. Though it would be nice to have answered either "do without me" or "wait for me" or whatever...

**Comment by defeo** [comment]
I'm sorry. If this is urgent, do without me. When I have time to get back to this, I'll eventually open another ticket if I see any problem.
